core/src/main/scala/kafka/server/AbstractFetcherThread.scala

```diff
@@ -52,6 +52,7 @@ import scala.math._
 abstract class AbstractFetcherThread(name: String,
                                      clientId: String,
                                      val sourceBroker: BrokerEndPoint,
+                                     markPartitionFailed: TopicPartition => Unit,
```

Comment:
I guess we could pass through `FailedPartitions` directly instead of this callback. The only thing we wouldn't have is the call to `shutdownIdleFetcherThreads`. But that is triggered periodically anyway, so maybe we don't ned to trigger it eagerly.

Reply:
Yeah, right. Modified the arguments to now accept the instance of `FailedPartitions` class.